keps/3122-expose-flavors-in-localqueue-status/README.md

```diff
@@ -0,0 +1,158 @@
+# KEP-3122: Expose Flavors in LocalQueue Status
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+  - [Implementation overview](#implementation-overview)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This KEP introduces a new status field in LocalQueue, allowing users to see
+all currently available ResourceFlavors in the LocalQueue.
+
+## Motivation
+
+Currently, users without RBAC access to ResourceFlavors cannot view the list
+of available flavors. Depending on the RBAC rules, users might also lack read
+access to ClusterQueues. Providing users with information about the available
+flavors is useful, as it gives them an idea of the capabilities provided by
+a LocalQueue (e.g., a flavor might include newer GPUs).
+
+### Goals
+
+- Provide a possibility to see all currently available ResourceFlavors in 
+  the LocalQueue.
+
+### Non-Goals
+
+- Verify that the ResourceFlavors exist and show only the existing flavors.
+
+## Proposal
+
+Introduce a new status field `availableFlavors` in LocalQueue 
+that will be updated when ClusterQueue flavors are modified.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user I want to see the list of all ResourceFlavors available in each LocalQueue 
+due to the RBAC configuration for ClusterQueue in my cluster I cannot inspect the 
+ClusterQueue objects directly, only LocalQueues.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+### API
+
+Create `AvailableFlavor` API object:
+
+```go
+type AvailableFlavor struct {
+  // name of the flavor.
+  Name ResourceFlavorReference `json:"name"`
+  // resources used in the flavor.
+  // +listType=set
+  // +optional
+  Resources []string `json:"resources,omitempty"`
+}
+```
+
+Modify `LocalQueueStatus` API object:
+
+```go
+// LocalQueueStatus defines the observed state of LocalQueue
+type LocalQueueStatus struct {
+	...
+	// availableFlavors lists all currently available ResourceFlavors
+	// in specified ClusterQueue.
+	//
+	// +listType=map
+	// +listMapKey=name
+	AvailableFlavors []AvailableFlavor `json:"availableFlavors,omitempty"`
+}
+```
+
+### Implementation overview
```

Comment:
What happens if the localQueue or tied ClusterQueue stopPolicy is changed?
For example, what if the tied ClusterQueue stopPlolicy is replaced with "Hold" or "HoldAndDrain"?

Reply:
It should happens nothing because on update StopPolicy we still keep it on cache.

Reply:
I'm wondering if we should clear / remove the availableFlavors when the cq or lq stopPolicy is Hold or HoldAndDrain since the flavor actually is not available.

@mimowo Any thoughts?

Reply:
I think it is ok to still show `availableResources` even if the LQ is on hold. IIUC for some use cases users may still want to submit workloads to a queue which is on hold. The hold can be cancelled later by the admin.
I think the fact that the queue is on hold can be better conveyed by other status fields or conditions.

Reply:
> EDIT: maybe we can make it less confusing by naming just flavors, or referencedFlavors?

The `flavors` sounds nice to me. Indeed, we have some resourceFlavor referenced field in the ClusterQueue status as `flavorsUsage` and `flavorsReservation` instead of `referencedFlavorsUsage`.
WDYT?

Reply:
Simple `flavors` is enough IMO.

Reply:
Awesome. Let's use the `flavors` as a field name.
@mbobrovskyi Additionally, could you mention that the stopPolicies are never considered in the lq flavors status field?

Reply:
Added to non-goals [1bb9e0f](https://github.com/kubernetes-sigs/kueue/pull/3181/commits/1bb9e0fc610267ba4a0de968fb48e80a75152ab4).